service/history/ndc/replication_task.go

```diff
@@ -127,7 +127,10 @@ func newReplicationTask(
 	lastEvent := events[len(events)-1]
 	version := firstEvent.GetVersion()
 
-	sourceCluster := clusterMetadata.ClusterNameForFailoverVersion(true, version)
+	sourceCluster, err := clusterMetadata.ClusterNameForFailoverVersion(true, version)
+	if err != nil {
+		return nil, err
```

Comment:
This will fail the entire get replication task call or just drop this replication task?

Reply:
It will fail the apply replication task call and the replication stack will put this task into DLQ after some retries.